Data/ByteString.hs

```diff
@@ -1,6 +1,7 @@
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE MagicHash, UnboxedTuples,
-            NamedFieldPuns, BangPatterns #-}
+            NamedFieldPuns, BangPatterns,
+            MultiWayIf #-}
```

Comment:
please avoid the non-criticial use of `MultiWayIf`

you can use  the Haskell2010 compliant pattern below if you need more than 2 branches

```
case () of
   _ | cond1 -> ...
      | cond2 -> ...
      | otherwise -> ...
```